### PR TITLE
feat(dbt): use `DbtProject` in `dagster-dbt project scaffold --use-experimental-dbt-project`

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary & Motivation
When explicit opt-in is given, use `DbtProject` in `dagster-dbt project scaffold`.

## How I Tested These Changes
pytest

### CODE REVIEW

**Comment by rexledesma** [comment]
* **#21089** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21089?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rexledesma and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/libraries/dagster-dbt/dagster_dbt/cli/app.py [hunk 1]
```diff
@@ -227,10 +233,12 @@ def project_scaffold_command(
...
     ] = False,
-    use_dbt_project_package_data_dir: Annotated[
+    use_experimental_dbt_project: Annotated[
         bool,
         typer.Option(
-            default=...,
+            ...,
+            "--use-dbt-project-package-data-dir",
+            "--use-experimental-dbt-project",
             help="Controls whether the dbt project package data directory is used.",
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/cli/app.py:242]: [1]

File: python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/project.py.jinja [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from pathlib import Path
+
+from dagster_dbt import DbtProject
+
+{{ dbt_project_name }} = DbtProject(
+    project_dir=Path(__file__).joinpath({{ dbt_project_dir_relative_path_parts | join(', ')}}).resolve(),
+    packaged_project_dir=Path(__file__).joinpath("..", "..", "dbt-project").resolve(),
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/project.py.jinja:7]: [1] its not clear to me that wiring up `--use-experimental-dbt-project` to also do the package data stuff is right. I am not sure how negatively the package data steps will impact users who do not intend for it, but it seems worth avoiding 

I think we want
* `--use-dbt-project-package-data-dir` turns on `DbtProject` and package data
* `--use-experimental-dbt-project` only turns on `DbtProject`

**Review by alangenfeld — CHANGES_REQUESTED** [review]
theres a `use_dbt_project_package_data_dir` ref in `python_modules/libraries/dagster-dbt/dagster_dbt/include/setup.py.jinja` too that may need updating

File: python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/project.py.jinja (hunk 2 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/include/scaffold/project.py.jinja:8]: Spoke offline. Going to do the following in this PR:

- `“--use-experimental-dbt-project”`: Scaffold using DbtProject
- `“--use-dbt-project-package-data-dir”`: Scaffold using DbtProject with its `packaged_project_dir` argument populated.

In a follow-up PR, I'll try to consolidate these two flags by making `packaged_project_dir` create a symlink during local dev.

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 3]
```diff
@@ -66,7 +66,7 @@ def prepare(self, project: "DbtProject") -> None:
             DbtCliResource(project_dir=os.fspath(project.project_dir))
             .cli(
                 self._generate_cli_args,
-                target_path=project.target_dir,
+                target_path=project.manifest_path.parent,
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:69]: Why this change? This should respect `target_dir` still right? 

The expectation in the package data flow is that we run prepare in to the source folder first and then sync it to package data second.

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 4]
```diff
@@ -116,10 +116,10 @@ def __init__(
             raise DagsterDbtProjectNotFoundError(f"project_dir {project_dir} does not exist.")
 
         packaged_project_dir = Path(packaged_project_dir) if packaged_project_dir else None
-        if using_dagster_dev() and packaged_project_dir:
+
+        current_project_dir = project_dir
+        if packaged_project_dir and packaged_project_dir.exists():
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:121]: I had this gated by `using_dagster_dev` before to ensure that we were always pointed at the source of truth project directory and not a potentially out of sync package data copy.

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_scaffold.py [hunk 5]
```diff
@@ -111,23 +123,26 @@ def test_project_scaffold_command_with_precompiled_manifest(
...
 
     _assert_scaffold_invocation(
         project_name=project_name,
         dbt_project_dir=dbt_project_dir,
         dagster_project_dir=dagster_project_dir,
         use_dbt_project_package_data_dir=use_dbt_project_package_data_dir,
+        use_experimental_dbt_project=use_experimental_dbt_project,
     )
 
     dbt_project_dir = _update_dbt_project_path(
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_scaffold.py:148]: pretty sure this should get replaced with prepare-for-deployment, check https://github.com/dagster-io/dagster/pull/19925/files for reference

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py [hunk 6]
```diff
@@ -116,10 +116,10 @@ def __init__(
...
             current_project_dir = packaged_project_dir
-        else:
-            current_project_dir = project_dir
 
         manifest_path = current_project_dir.joinpath(target_dir, "manifest.json")
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:125]: `project_dir=project_dir,` may need to get updated to `current_project_dir` - the package data stuff isn't really under test until this PR

File: python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_scaffold.py (hunk 5 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_scaffold.py:148]: I just pushed up [python_modules/libraries/dagster-dbt/dagster_dbt_tests/cli/test_prepare_for_deployment.py](https://github.com/dagster-io/dagster/pull/21089/files#diff-6253b51694e1d8c24fbe934f0b1badc12826b15ddffeef78b68d2bac774ac285), which will test that flow.

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py (hunk 4 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:121]: Hmm, shouldn't the logic then be:

```python
current_project_dir = project_dir
if not using_dagster_dev() and packaged_project_dir and packaged_project_dir.exists():
    current_project_dir = packaged_project_dir
```

File: python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py (hunk 3 above)
Comment (rexledesma) [reply python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py:69]: We should be able to revert this once we set `project_dir=current_project_dir`

**Comment by rexledesma** [comment]
@alangenfeld Re: 

> the exists check - not sure, if you are expecting it to be in package data but you didn't make package data dir I think you want an error that points at that directory? Maybe an explicit not exists -> error path is the way

I don't think we can do an explicit error here, otherwise, we would fail when we try to call `dagster-dbt project prepare-for-deployment`.

Maybe we could introduce another variable here `DAGSTER_DBT_IS_PREPARE_FOR_DEPLOYMENT` so that we can skip that exception if we're not in a context that is building the packaged project dir?
